src/mlpack/methods/preprocess/preprocess_string_encoding_main.cpp

```diff
@@ -0,0 +1,234 @@
+ */
+
+#include "mlpack/methods/preprocess/preprocess_string_util.cpp"
+
+PROGRAM_INFO("preprocess_string_encoding",
+    // Short description.
+    "A utility to encode string data. This utility can encode string using "
+    "DictionaryEncoding, BagOfWordsEncoding and TfIdfEncoding methods.",
+    // Long description.
+    "This utility takes a dataset and the dimension and arguments and "
+    "encodes string dataset according to arguments given."
+    "\n\n"
+    "The dataset may be given as the file name and the output may be saved as "
+    + PRINT_PARAM_STRING("actual_dataset") + " and " +
+    PRINT_PARAM_STRING("preprocess_dataset") + " ."
+    "\n\n"
+    " Following arguments may be given " + PRINT_PARAM_STRING("encoding_type") +
+    " to encode the dataset using a specific encoding type and " + " Also the "
+    "dimension which contains the string dataset " +
+    PRINT_PARAM_STRING("dimension") + "."
+    "\n\n"
+    "So, a simple example where we want to encode string dataset " +
+    PRINT_DATASET("X") + ", which is having string data in its 3 Column,"
+    " using DictionaryEncoding as encoding type."
+    "\n\n" +
+    PRINT_CALL("preprocess_string", "actual_dataset", "X",
+        "preprocess_dataset", "X", "dimension", 3, "encoding_type",
+        "DictionaryEncoding") +
+    "\n\n",
+    SEE_ALSO("@preprocess_binarize", "#preprocess_binarize"),
+    SEE_ALSO("@preprocess_describe", "#preprocess_describe"),
+    SEE_ALSO("@preprocess_imputer", "#preprocess_imputer"));
+
+PARAM_STRING_IN_REQ("actual_dataset", "File containing the reference dataset.",
+    "t");
+PARAM_STRING_IN_REQ("preprocess_dataset", "File containing the preprocess or"
+    "encoded dataset.", "o");
+PARAM_STRING_IN("column_delimiter", "delimeter used to seperate Column in files"
+    "example '\\t' for '.tsv' and ',' for '.csv'.", "d", "\t");
+PARAM_STRING_IN("delimiter", "A set of chars that is used as delimeter to"
+    "tokenize the string dataset ", "D", " ");
+PARAM_STRING_IN("encoding_type","Type of encoding","e","DictionaryEncoding")
+PARAM_STRING_IN("tfidf_encoding_type","Type of tfidf encoding","E","RawCount")
+PARAM_FLAG("smooth_idf", "True to have smooth_idf for Tf-Idf.", "s");
+PARAM_VECTOR_IN_REQ(std::string, "dimension", "Column which contains the"
+    "string data. (1 by default)", "c");
+
+using namespace mlpack;
+using namespace mlpack::util;
+using namespace arma;
+using namespace std;
+
+/**
+ * Function used to write back the preproccessed data to a file.
+ *
+ * @param outputFilename Name of the file to save the data into.
+ * @param dataset The actual dataset which was read from file.
+ * @param columnDelimiter Delimiter used to separate columns of the file.
+ * @param dimesnions Dimesnion which we non numeric or of type string.
+ * @param encodedResult Collection of arma matrices havinf encoded Results.
+ */
+static void WriteOutput(const string& outputFilename,
```

Comment:
Why didn't you move this function to a separate file?

Reply:
It is not same in both binding, That's why.